runtime/domain.c

```diff
@@ -935,13 +936,29 @@ void caml_update_minor_heap_max(uintnat requested_wsz) {
   check_minor_heap();
 }
 
-void caml_init_domains(uintnat minor_heap_wsz) {
+void caml_init_domains(uintnat minor_heap_wsz, uintnat max_domains) {
   int i;
 
+  all_domains = caml_stat_alloc_noexc(max_domains * sizeof(dom_internal));
```

Comment:
```suggestion
  all_domains = caml_stat_calloc_noexc(max_domains * sizeof(dom_internal));
```
Users inside `caml_try_run_on_all_domains_with_spin_work` and `caml_interrupt_all_signal_safe` expect it to be zero-initialised.

Reply:
Done in https://github.com/ocaml/ocaml/pull/13272/commits/263be361baad53f6ef36f8774ac8b5d9b1572525.